### PR TITLE
Use ctc lib for container structure tests

### PR DESCRIPTION
**Pull request by tejal29**
In this change
1. Add runtimes-common/ctc_lib as a dependency.
2. Remove version command
3. Change all the RunAll and Run method of Structure Tests to write to stream
4. Add default template
5. Finally add a aggregate function Total and define template for Aggregation.
6. Get rid of --verbose flag and use --quiet instead.
7. Change all logrus to ctc_lib.Log.

### CODE REVIEW

File: cmd/test.go [hunk 1]
```diff
@@ -43,23 +44,60 @@ var totalTests int
...
+	TotalO: func(list []interface{}) (interface{}, error) {
+		totalPass := 0
+		totalFail := 0
+		errStrings := make([]string, 0)
+		var err error = nil
+		for _, r := range list {
+			value, ok := r.(*unversioned.TestResult)
+			if !ok {
+				errStrings = append(errStrings, fmt.Sprintf("UnExpected Value %v in List.", value))
+				ctc_lib.Log.Errorf("UnExpected Value %v in List.", value)
```
Comment (nkubala) [inline cmd/test.go]: nit: `UnExpected Value` -> `Unexpected value`

File: cmd/test.go [hunk 2]
```diff
@@ -173,23 +211,24 @@ func Run(out *output.OutWriter) bool {
...
 		}, docker.AuthConfiguration{}); err != nil {
-			logrus.Fatalf("Error pulling remote image %s: %s", imagePath, err.Error())
+			ctc_lib.Log.Fatalf("Error pulling remote image %s: %s", imagePath, err.Error())
 		}
 	}
 
 	if driver == drivers.Host && !utils.UserConfirmation(warnMessage, force) {
-		return false
+		// Force Success. Skipping Running tests as directed by the user.
+		return
```
Comment (nkubala) [inline cmd/test.go]: I think we might want to force failure here? Similar to how ctrl+c usually causes applications to exit 1.

File: pkg/types/v1/command.go [hunk 3]
```diff
@@ -95,30 +95,30 @@ func (ct *CommandTest) LogName() string {
 func (ct *CommandTest) CheckOutput(result *types.TestResult, stdout string, stderr string, exitCode int) {
 	for _, errStr := range ct.ExpectedError {
 		if !utils.CompileAndRunRegex(errStr, stderr, true) {
-			result.Errorf("Expected string '%s' not found in error", errStr)
-			result.Fail()
+			// result.Errorf("Expected string '%s' not found in error", errStr)
```
Comment (nkubala) [inline pkg/types/v1/command.go]: uncomment these

File: pkg/types/v1/file_existence.go [hunk 4]
```diff
@@ -51,7 +51,7 @@ func (ft FileExistenceTest) Run(driver drivers.Driver) *types.TestResult {
 		Pass:   true,
 		Errors: make([]string, 0),
 	}
-	logrus.Info(ft.LogName())
+
```
Comment (nkubala) [inline pkg/types/v1/file_existence.go]: `ctc_lib.Log.Info(ft.LogName())`?

File: cmd/test.go [hunk 5]
```diff
@@ -199,8 +238,22 @@ func init() {
...
-	TestCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "verbose testing output")
+	TestCmd.Flags().BoolVar(&force, "force", false, "force run of host driver (without user prompt)")
+
 	TestCmd.Flags().StringArrayVar(&configFiles, "config", []string{}, "test config files")
 }
+
+func isQuiet() bool {
+	return quiet
+}
+func initTemplateFuncMap() map[string]interface{} {
```
Comment (nkubala) [inline cmd/test.go]: what's this for?

File: pkg/drivers/docker_driver.go [hunk 6]
```diff
@@ -93,10 +94,10 @@ func (d *DockerDriver) ProcessCommand(envVars []unversioned.EnvVar, fullCommand
 	}
 
 	if stdout != "" {
-		logrus.Infof("stdout: %s", stdout)
+		ctc_lib.Log.Debugf("stdout: %s", stdout)
```
Comment (nkubala) [inline pkg/drivers/docker_driver.go]: can these stay info level?

File: cmd/test.go (hunk 2 above)
Comment (tejal29) [reply cmd/test.go]: umm, i am of opinion that if user pressed 'n'  it not an tool error as the tool did what the user asked him to do.

File: cmd/test.go (hunk 5 above)
Comment (tejal29) [reply cmd/test.go]: This is to make the Functions available inside the template.
https://golang.org/pkg/text/template/#FuncMap

File: pkg/drivers/docker_driver.go (hunk 6 above)
Comment (tejal29) [reply pkg/drivers/docker_driver.go]: sure.

**Comment by tejal29** [comment]
There are no changes to the result.
1. On Branch 
https://drive.google.com/file/d/1rjZnJoLtEMZDbp3O0iRtjCe9RLlIjei9/view?usp=sharing

2. On master
https://drive.google.com/file/d/1N0lazg68FoQnZslgvQBtNbMw7-Wrvpmf/view?usp=sharing

File: pkg/types/v1/file_existence.go (hunk 4 above)
Comment (tejal29) [reply pkg/types/v1/file_existence.go]: done

File: pkg/types/v1/command.go (hunk 3 above)
Comment (tejal29) [reply pkg/types/v1/command.go]: done

File: cmd/test.go (hunk 1 above)
Comment (tejal29) [reply cmd/test.go]: done

File: cmd/test.go (hunk 2 above)
Comment (tejal29) [reply cmd/test.go]: I changed my mind and error out "User Aborted" Message

File: cmd/test.go [hunk 7]
```diff
@@ -43,23 +44,60 @@ var totalTests int
...
+	Stream: Channel,
+	TotalO: func(list []interface{}) (interface{}, error) {
+		totalPass := 0
+		totalFail := 0
+		errStrings := make([]string, 0)
+		var err error = nil
+		for _, r := range list {
+			value, ok := r.(*unversioned.TestResult)
+			if !ok {
+				errStrings = append(errStrings, fmt.Sprintf("UnExpected Value %v in List.", value))
```
Comment (nkubala) [inline cmd/test.go]: nit: `UnExpected Value` -> `Unexpected value`

**Pull request merged**